lib/utils/useWeb3Modal/index.tsx

```diff
@@ -29,52 +28,112 @@ export const useWeb3Modal = (strings: Web3ModalStrings): Web3ModalState => {
     }
   };
 
-  const downloadModal = async () => {
-    const newModal = (await import("../getWeb3Modal")).getWeb3Modal(strings);
-    setPrevStrings(strings);
-    return newModal;
-  };
+  const connect = async (wallet?: string): Promise<void> => {
+    try {
+      if (wallet === "metamask" || wallet === "brave" || window.ethereum) {
+        const provider = new ethers.providers.Web3Provider(
+          window.ethereum as any,
+          137
+        ) as any;
+        await provider.send("eth_requestAccounts", []);
```

Comment:
what does this do?

Reply:
added comment `        // if user is not already connected this request will prompt the wallet modal to open and the user to connect
`